src/meta/processors/admin/AdminClient.cpp

```diff
@@ -335,6 +336,51 @@ StatusOr<std::vector<HostAddr>> AdminClient::getPeers(GraphSpaceID spaceId, Part
     return Status::Error("Get Failed");
 }
 
+folly::Future<Status> AdminClient::getLeaderDist(HostLeaderMap* result) {
+    if (injector_) {
+        return injector_->getLeaderDist(result);
+    }
+    folly::Promise<Status> promise;
+    auto future = promise.getFuture();
+    auto allHosts = ActiveHostsMan::getActiveHosts(kv_);
+
+    auto getLeader = [result, this] (const HostAddr& host) {
+        storage::cpp2::GetLeaderReq req;
+        folly::Promise<Status> pro;
+        auto f = pro.getFuture();
+        auto* evb = ioThreadPool_->getEventBase();
+        auto client = clientsMan_->client(host, evb);
```

Comment:
we'd better run client->future_getLeaderPart inside evb,  because it will rebuild the connection if lose.  And the procedure need to be invoked inside ioThread

Reply:
Got it.